sale_timesheet_invoice_description/models/sale.py

```diff
@@ -47,9 +52,14 @@ def _prepare_invoice_line(self, qty):
         domain = [('so_line', '=', self.id)]
         last_invoice = self.invoice_lines.sorted(lambda x: x.create_date)[-1:]
         if last_invoice:
-            domain.append(('create_date', '>', last_invoice.create_date))
-        for line in self.env['account.analytic.line'].search(
-                domain, order='date, id'):
+            last_date = fields.Datetime.to_string(last_invoice.create_date)
+            domain.append(('create_date', '>', last_date))
+
+        # If project_timesheet_time_control module is installed,
+        # date field is hidden and the new date_time field is used instead
+        line_obj = self.env['account.analytic.line']
+        order = hasattr(line_obj, 'date_time') and 'date_time' or 'date'
+        for line in line_obj.search(domain, order=order + ', id'):
```

Comment:
Why don't rely in default order set by the model instead (and more now that we are going to set it properly)?

Reply:
I think you're right, thank you